db/migrations/20190711091453_create_flip_table.sql

```diff
@@ -22,102 +23,132 @@ CREATE INDEX flip_address_index
+  AND flip_bid_tic.address_id = flip_bid_tic_before_block.address_id
+  AND headers.block_number < (SELECT block_number FROM public.headers WHERE id = flip_bid_tic_before_block.header_id)
 ORDER BY block_number DESC
 LIMIT 1
 $$
     LANGUAGE sql;
 
-COMMENT ON FUNCTION get_latest_flip_bid_tic
+COMMENT ON FUNCTION flip_bid_tic_before_block
     IS E'@omit';
 
-CREATE FUNCTION get_latest_flip_bid_end(bid_id numeric) RETURNS BIGINT AS
+CREATE FUNCTION flip_bid_end_before_block(bid_id NUMERIC, address_id INTEGER, header_id INTEGER) RETURNS BIGINT AS
 $$
 SELECT "end"
-FROM maker.flip
-WHERE "end" IS NOT NULL
-  AND flip.bid_id = bid_id
+FROM maker.flip_bid_end
+         LEFT JOIN public.headers ON flip_bid_end.header_id = headers.id
+WHERE flip_bid_end.bid_id = flip_bid_end_before_block.bid_id
+  AND flip_bid_end.address_id = flip_bid_end_before_block.address_id
+  AND headers.block_number < (SELECT block_number FROM public.headers WHERE id = flip_bid_end_before_block.header_id)
 ORDER BY block_number DESC
 LIMIT 1
 $$
     LANGUAGE sql;
 
-COMMENT ON FUNCTION get_latest_flip_bid_end
+COMMENT ON FUNCTION flip_bid_end_before_block
     IS E'@omit';
 
-CREATE FUNCTION get_latest_flip_bid_lot(bid_id numeric) RETURNS NUMERIC AS
+CREATE FUNCTION flip_bid_lot_before_block(bid_id NUMERIC, address_id INTEGER, header_id INTEGER) RETURNS NUMERIC AS
 $$
 SELECT lot
-FROM maker.flip
-WHERE lot IS NOT NULL
-  AND flip.bid_id = bid_id
+FROM maker.flip_bid_lot
+         LEFT JOIN public.headers ON flip_bid_lot.header_id = headers.id
+WHERE flip_bid_lot.bid_id = flip_bid_lot_before_block.bid_id
+  AND flip_bid_lot.address_id = flip_bid_lot_before_block.address_id
+  AND headers.block_number < (SELECT block_number FROM public.headers WHERE id = flip_bid_lot_before_block.header_id)
 ORDER BY block_number DESC
 LIMIT 1
 $$
     LANGUAGE sql;
 
-COMMENT ON FUNCTION get_latest_flip_bid_lot
+COMMENT ON FUNCTION flip_bid_lot_before_block
     IS E'@omit';
 
-CREATE FUNCTION get_latest_flip_bid_bid(bid_id numeric) RETURNS NUMERIC AS
+CREATE FUNCTION flip_bid_bid_before_block(bid_id NUMERIC, address_id INTEGER, header_id INTEGER) RETURNS NUMERIC AS
 $$
 SELECT bid
-FROM maker.flip
-WHERE bid IS NOT NULL
-  AND flip.bid_id = bid_id
+FROM maker.flip_bid_bid
+         LEFT JOIN public.headers ON flip_bid_bid.header_id = headers.id
+WHERE flip_bid_bid.bid_id = flip_bid_bid_before_block.bid_id
+  AND flip_bid_bid.address_id = flip_bid_bid_before_block.address_id
+  AND headers.block_number < (SELECT block_number FROM public.headers WHERE id = flip_bid_bid_before_block.header_id)
 ORDER BY block_number DESC
 LIMIT 1
 $$
     LANGUAGE sql;
 
-COMMENT ON FUNCTION get_latest_flip_bid_bid
+COMMENT ON FUNCTION flip_bid_bid_before_block
     IS E'@omit';
 
-CREATE FUNCTION get_latest_flip_bid_gal(bid_id numeric) RETURNS TEXT AS
+CREATE FUNCTION flip_bid_usr_before_block(bid_id NUMERIC, address_id INTEGER, header_id INTEGER) RETURNS TEXT AS
+$$
+SELECT usr
```

Comment:
Nice catch, I probably should've explained why I added this! The short answer is no, this isn't a new field. 

The `usr` field isn't in the spec, and tbh I added it on a whim because it was very easy and I wanted the flip table's structure to match [the `Bid` object](https://github.com/makerdao/dss/blob/master/src/flip.sol#L52) in the solidity code. The only real benefit I see to keeping it, is that the `usr` is an urn identifier, so having it directly available in this table will make the `urn` computed column more performant, since it won't have to do a separate lookup to find the `usr` associated with a bid.